### PR TITLE
Tracker/MTD geometry: clean Geometric(Timing)Det dependence on DDD

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

Following the discussion in #28914 , I propose a cleaning of ```GeometricDet``` and ```GeometricTimingDet``` classes to remove the dependence on ```DDExpandedNode```, which as far as I can see is just fake. In order to make things transparent, I add in ```DDCMS``` a few ```using``` statements included in ```DDD``` which may help in making transparent the migration. 

A non trivial dependence remains in ```GeometricDetExtras``` and into its timing counterpart. As far as I can see:

- the use of ```DDExpandedNode``` is real, and a fully independent implementation of similar functionalities should be made using DD4hep if needed, so far there is none;

- this is anyway not impacting the persistent ```PGeometricDet```;

- the use of this object is for pure debugging purpose only, as far as I can see, so in principle its removal should not break production workflows. For sure this is true for ```GeometricTimingDetExtras``` that appears to be a cut-and-paste minimal adjustment of the tracker code, never used and not providing meaningful results at present as far as I can see.

For this reason I propose here a simple removal of this part for MTD, to be possibly replaced with something else in future. But there are already debugging classes that can be possibly extended in case.

As far as Tracker geometry is concerned, it is up to the DPG experts say whether they still need these functionalities, and in case provide a DD4hep-based implementation (that could replace asymptotically the DDD implementation).

#### PR validation:

MTD geometry unit tests and wf 23234.0 run.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
hold

I apparently missed this issue in my second commit, checking it again

**Comment by fabiocos** [comment]
please test

File: Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc [hunk 1]
```diff
@@ -53,6 +53,8 @@
 #include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 #include "Geometry/Records/interface/TrackerTopologyRcd.h"
 
+#include "DetectorDescription/Core/interface/DDCompactView.h"
```
Comment (cvuosalo) [inline Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc:56]: Can you please explain this file? It is not migrated. Is it part of `GeometricTimingDetExtras`?

Comment (fabiocos) [reply Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc:56]: @cvuosalo in the case of this file, as well as the ```RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc``` it is a matter of nested include dependencies, that now are exposed by the removal of the ```DDExpandedView``` from ```GeometricDet```. The compilation fails as soon as you update that class.

**Comment by fabiocos** [comment]
unhold

File: Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc (hunk 1 above)
Comment (cvuosalo) [reply Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc:56]: My confusion results from two different versions of this file being included in #29905 and this PR. #29905 has the migrated version. If #29905 is merged first, this PR will probably need to be rebased.

File: RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include "Phase2EndcapRingBuilder.h"
 #include "TrackingTools/DetLayers/interface/DetLayerException.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
```
Comment (cvuosalo) [inline RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc:3]: Does this file even belong in this PR? It doesn't seem related, and it brings in the requirement for a Reco signature.

**Comment by cvuosalo** [comment]
+1

**Comment by cvuosalo** [comment]
The decision in this PR is to leave the Tracker version of `GeometricDetExtras` unmigrated. If the Tracker DPG wants to use `GeometricDetExtras` tool in the future, they will have to migrate it.

File: Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc (hunk 1 above)
Comment (fabiocos) [reply Alignment/OfflineValidation/plugins/TrackerGeometryCompare.cc:56]: @cvuosalo up to the release manager decide which PR is mature for integration first. In case I will rebase mine on that one, not a big issue. BTW, I see that @ianna has already made this addition in #29905

File: RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc (hunk 2 above)
Comment (fabiocos) [reply RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc:3]: @cvuosalo unfortunately yes, without it the code does not compile, you may easily try yourself.

**Comment by fabiocos** [comment]
@cvuosalo as I said, I believe that it is up to the Tracker DPG decide what they want to do with this object. For MTD it is not used so far, and its implementation is not really usable, just a cut-and-paste adjustment of the tracker code that is unusable in its present form. This is not true for the tracker original object, but a DD4hep based implementation of that requires some non completely trivial work, which goes beyond my main task related to MTD. I wonder to which extent this debugging code is really used nowadays, it looks useless to invest work in updating dead code (if that is the case).

**Comment by fabiocos** [comment]
to be even more explicit: this object is used in just 3 places in test modules as far as I can see, either through 

https://cmssdt.cern.ch/lxr/source/Geometry/TrackerNumberingBuilder/src/CmsTrackerDebugNavigator.cc#0017

or to print equivalent information, and the name of some higher level volumes in the stack. One should replace the equivalent information with something from DD4hep filtered view.  The information coming from DDD logical parts should be appropriately replaced, the copy number is easily accessible, as far as the GeoHistory is concerned, for what I can see most probably the information needed in https://cmssdt.cern.ch/lxr/source/SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc#0433 can be retrieved through the DD4hep geoHistory() as done in https://cmssdt.cern.ch/lxr/source/Geometry/MTDNumberingBuilder/test/DD4hep_MTDTopologyAnalyzer.cc#0210

**Comment by perrotta** [comment]
+1
- Just a technical fix for reco
- Jenkins tests pass

**Comment by pohsun** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1
